gax-grpc/src/test/java/com/google/api/gax/grpc/GrpcCallContextTest.java

```diff
@@ -210,4 +214,41 @@ public void testMergeWithCustomCallOptions() {
     Truth.assertThat(merged.getCallOptions().getOption(key))
         .isEqualTo(ctx2.getCallOptions().getOption(key));
   }
+
+  @Test
+  public void testWithExtraHeaders() {
+    Map<String, List<String>> extraHeaders = createTestExtraHeaders();
+    GrpcCallContext ctx = GrpcCallContext.createDefault();
+    ctx = ctx.withExtraHeaders(extraHeaders);
+    Map<String, List<String>> gotExtraHeaders = ctx.getExtraHeaders();
+    Truth.assertThat(gotExtraHeaders).isNotSameAs(extraHeaders);
+    Truth.assertThat(gotExtraHeaders)
+        .containsEntry(
+            "header-key-1", ImmutableList.<String>of("header-value-11", "header-value-12"));
+    Truth.assertThat(gotExtraHeaders)
+        .containsEntry("header-key-2", ImmutableList.<String>of("header-value-21"));
+  }
+
+  @Test
+  public void testMergeWithExtraHeaders() {
+    Map<String, List<String>> extraHeaders = createTestExtraHeaders();
+    GrpcCallContext ctx1 = GrpcCallContext.createDefault();
+    GrpcCallContext ctx2 = GrpcCallContext.createDefault().withExtraHeaders(extraHeaders);
+    ApiCallContext mergedApiCallContext = ctx1.merge(ctx2);
+    Truth.assertThat(mergedApiCallContext).isInstanceOf(GrpcCallContext.class);
+    GrpcCallContext mergedGrpcCallContext = (GrpcCallContext) mergedApiCallContext;
+    Truth.assertThat(mergedGrpcCallContext.getExtraHeaders())
```

Comment:
A later version of Truth has methods like `containsExactlyEntriesIn` that can make the tests more elegant. Shall we update to the latest or keep it as it is?

Reply:
That sounds neat - let's upgrade.